src/lib/plugin.js

```diff
@@ -29,73 +29,107 @@ module.exports = class PluginVirtual extends EventEmitter2 {
   constructor (opts) {
     super()
 
-    assertOptionType(opts, 'currencyCode', 'string')
-    assertOptionType(opts, 'currencyScale', 'number')
-    assertOptionType(opts, 'maxBalance', 'string')
-    assertOptionType(opts, 'secret', 'string')
-    assertOptionType(opts, 'peerPublicKey', 'string')
-    assertOptionType(opts, '_store', 'object')
-    assertOptionType(opts, 'rpcUri', 'string')
-
-    this._secret = opts.secret
-    this._peerPublicKey = opts.peerPublicKey
-    this._publicKey = Token.publicKey(this._secret)
-    this._currencyCode = opts.currencyCode.toUpperCase()
-    this._currencyScale = opts.currencyScale
+    this._stateful = !!opts._store
```

Comment:
I think the plugin should be stateless by default, because that's the way most people are going to use it in their code. The stateful plugins will always be created by some API. Furthermore, if you try to instantiate a stateless plugin without a corresponding stateful one, the connect step will fail (there won't be anyone to respond to the `get_info` RPC message)